pkg/datapath/types/loader.go

```diff
@@ -57,7 +57,7 @@ type Proxy interface {
 type IptablesManager interface {
 	// InstallProxyRules creates the necessary datapath config (e.g., iptables
 	// rules for redirecting host proxy traffic on a specific ProxyPort)
-	InstallProxyRules(proxyPort uint16, isLocalOnly bool, name string)
+	InstallProxyRules(proxyPort uint16, localOnly bool, name string) <-chan struct{}
```

Comment:
> Being needed just for backward compatibility with the tests suite, it
may be removed in the future when the tests are rewritten to be less
unforgiven.

:sweat_smile: 

So at runtime we need this to be async, while during testing it should be sync? Looking at the code paths in endpoint it seems like those also want to be sync? My gut feeling would be to just keep the old interface and hide the channel read.

If you decide to stick with this: Nit: document when the returned channel is closed?

Reply:
> So at runtime we need this to be async, while during testing it should be sync?

That's unfortunately the case. :cry: 
I've tried many other solutions to avoid performing synchronous operations but with no luck (see the CI runs in https://github.com/cilium/cilium/pull/31024, https://github.com/cilium/cilium/pull/30918, https://github.com/cilium/cilium/pull/30853 and https://github.com/cilium/cilium/pull/30821).
I've also failed to identify a reasonably small set of tests to update, that's why in the end I decided to go with this and finalize the PR. This refactor is also blocking the runtime devices reconciliation effort https://github.com/cilium/cilium/pull/30024

> Looking at the code paths in endpoint it seems like those also want to be sync? My gut feeling would be to just keep the old interface and hide the channel read.

On the one hand exposing the channel makes it clearer that the operation can (and should) be performed asynchronously, but on the other hand it is true that all the iptables consumers are written for the previous synchronous version.
Let's hide the channel as you suggested, it is probably better to keep it cleaner until we refactor the consumers (and the tests). :+1: